### DISCUSSION TITLE
Multiple API filters depending on conditions

### DISCUSSION THREAD

**Discussion by hypulse** [pocketbase/pocketbase#4425] [Q&A] [answered]
I created a posts collection with a select field for 'private' and 'public' options.

Using the JavaScript SDK, I implemented an API endpoint where logged-in users can see both 'private' and 'public' posts, while anonymous users can only see 'public' posts.

However, I found that direct API calls can't be prevented, and I need help with the correct syntax for implementing API rules.

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure that I understand what actually is the question.

Your requirement _'both "private" and "public" states to logged-in users, but only "private" to anonymous users'_ also confuses me - shouldn't anonymous users have access only to the "public" records?

>  there's no way to prevent direct calls to the server API

The JS SDK is a thin wrapper around `fetch` so practically there shouldn't be any differences in the response when making requests with the SDK or directly via something like curl.

In general, if you want to apply a conditional list or view API rule based on the logged auth state you can use the `||` (OR) operator:

```js
// for authenticated users - show everything no matter of the status
// for guests - show only those with status "private" (replace private to public depending on your use case)
@request.auth.id != "" || status = "private"
```

If the above doesn't help, please elaborate on the desired behavior and export your collections configuration as json (_Admin UI > Settings > Export collections_).
